File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.py
class Solution(object):
    def rightSideView(self, root):
        res = []

        def dfs(node, depth):
            if not node:
                return

            # 현재 depth에 처음 도달한 노드면 추가
            if depth == len(res):
                res.append(node.val)

            # 먼저 오른쪽 방문
            dfs(node.right, depth + 1)
            dfs(node.left, depth + 1)

        dfs(root, 0)
        return res

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
import java.util.*;

class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> answer = new ArrayList<>();
        if (root == null) {
            return answer;
        }

        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(root);

        while (!queue.isEmpty()) {
            int size = queue.size();

            for (int i = 0; i < size; i++) {
                TreeNode current = queue.poll();

                if (current.left != null) {
                    queue.offer(current.left);
                }
                if (current.right != null) {
                    queue.offer(current.right);
                }

                if (i == size - 1) {
                    answer.add(current.val);
                }
            }
        }

        return answer;
    }
}